lib/coverband/utils/file_path_helper.rb

```diff
@@ -8,16 +8,23 @@ module Utils
     module FilePathHelper
       module_function
 
+      @@path_cache = {}
+
       ###
       # Takes a full path and converts to a relative path
       ###
       def full_path_to_relative(full_path)
+        return @@path_cache[full_path] if @@path_cache.key?(full_path)
+
         relative_filename = full_path
-        Coverband.configuration.all_root_paths.each do |root|
-          relative_filename = relative_filename.gsub(/^#{root}/, './')
+        Coverband.configuration.all_root_patterns.each do |root|
+          relative_filename = relative_filename.sub(root, './')
```

Comment:
I wonder if it would help memory to use `sub!`

Reply:
yeah probably would... see this line https://github.com/danmayer/coverband/pull/290/files#diff-2d63c89a82b59b5370220c4697cdbee2R133

basically, I think we have a larger cleanup around all in memory edits or all reduction edits...